tracing/tracing/src/main/java/io/helidon/tracing/SpanLifeCycleListener.java

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * Behavior of listeners notified of span life cycle events.
+ * <p>
+ * The interface declares default empty implementations for each method, allowing concrete implementations of the
+ * interface to implement only the methods that are relevant for them.
+ * <p>
+ * Helidon invokes the applicable methods of a life cycle listener in the following order:
+ * <table>
+ *     <caption style="text-align:left">Order of Invocation of Listener Methods</caption>
+ *
+ *     <tbody>
+ *         <tr>
+ *         <th style="text-align:left;width:15%">Method</th>
+ *         <th style="text-align:left">When invoked</th>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #beforeStart(io.helidon.tracing.Span.Builder) beforeStart}</td>
+ *         <td>Before a span is started from its builder.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #afterStart(io.helidon.tracing.Span) afterStart}</td>
+ *         <td>After a span has started.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #afterActivate(io.helidon.tracing.Span, io.helidon.tracing.Scope)
+ *         afterActivate} †</td>
+ *         <td>After a span has been activated, creating a new scope in the process. </td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #afterClose(io.helidon.tracing.Span, io.helidon.tracing.Scope) afterClose} †</td>
+ *         <td>After a scope has been closed.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #afterEnd(io.helidon.tracing.Span) afterEnd (successful)} *</td>
+ *         <td>After a span has ended successfully.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>{@linkplain #afterEnd(io.helidon.tracing.Span, Throwable) afterEnd (unsuccessful)} *</td>
+ *         <td>After a span has ended unsuccessfully. </td>
+ *     </tr>
+ *     </tbody>
+ *      <tfoot>
+ *         <tr>
+ *             <td colspan="2">† Not all spans are activated; it is up to the application or library code that creates and manages the span.
+ *              As a result Helidon might not invoke your listener's {@code afterActivate} and {@code afterClose} methods for
+ *              every span.
+ *              <p>
+ *              * The successful or unsuccessful nature of a span's end is not about whether the tracing or telemetry
+ *              system failed to end the span. Rather, it indicates whether the code that ended the span indicated some error in
+ *              the processing which the span represents.
+ *             </td>
+ *         </tr>
+ *     </tfoot>
+ * </table>
+ * <p>
+ *     When Helidon invokes the listener methods it passes implementations of the key span types ({@link Span.Builder},
+ *     {@link Span}, {@link io.helidon.tracing.Scope}) which <em>do not</em> support lifecycle state changes.
+ *     If a lifecycle listener tries to start or end or activate a span,
+ *     for example, Helidon throws an {@link java.lang.UnsupportedOperationException}.
+ */
+public interface SpanLifeCycleListener {
```

Comment:
Rename to either `SpanListener` (preferred by your humble reviewer) or `SpanLifecycleListener`

Reply:
Changed